lib/bowling_game.rb

```diff
@@ -0,0 +1,29 @@
+class BowlingGame
+
+    def initialize
+        @rolls = []
+    end
+
+    def roll pins 
+        @rolls.push pins
+    end
+
+    def score
+        result = 0
+        rollIndex = 0
+        10.times do
+          if @rolls[rollIndex] + @rolls[rollIndex + 1] == 10
+            result += @rolls[rollIndex] + @rolls[rollIndex + 1] = @rolls[rollIndex + 2]
+          else
+            result += @rolls[rollIndex] + @rolls[rollIndex + 1]
+          end
+          rollIndex += 2
+        end
+        result
+    end
```

Comment:
Would love to know more about how you got this to work - I tried to use the same method but no matter what I did, I couldn't get the 'rollIndex' part to work! This is a really nice :)

Reply:
If you'd had more time, you could have refactored this a bit by breaking it up into new methods like 'spare?', 'strike?' etc - looks like you started doing that below!